src/com/stanimiriliev/chaptertwo/auto/Test.java

```diff
@@ -0,0 +1,23 @@
+package com.stanimiriliev.chaptertwo.auto;
+
+import com.stanimiriliev.chaptertwo.auto.Automobile.Color;
+
+import java.util.Date;
+
+/**
+ * Test functionality of the classes VW, Mercedes and Opel
+ * @author Stanimir Iliev
+ * @since 22.07.2017
+ */
+public class Test {
+    public static void main(String[] args){
+
+        VW golf = new VW(180, Color.BLUE, new Date());
+        Mercedes eClass = new Mercedes(240, Color.BLACK, new Date());
+        
```

Comment:
```java
Opel astra = new Opel(190, Color.GREY, new Date(), 1100, 5);
Automobile vectra = new Opel(200, Color.BLACK, new Date(), 1240, 5);
```
you can do this ;

```java
vectra.weight
vectra.passengers
```
bu you cannot do this with astra
```java
astra.weight
astra.passengers
```
because "astra" is declared as Automobile in your application, it is important how you define the object.
it is the same as the homogeneous tree you declare the interface, then you pass objects that implements the interface, but in the tree logic you actually use only the methods of the interface, because you do not have which object actually is passed - Milk or Bread it is important to access their comparable behavior.